### PR TITLE
Set the libraries of gpfdist and gpmapreduce at the top level

### PR DESCRIPTION
**Pull request by adam8157**
Before this the top level configure only sets the have_yaml, but doesn't
add -lyaml to LIBS for gpfdist or gpmapreduce.

Let Autoconf resolve all the relevant libraries here and place them in
$XXX_LIBS variables.

Signed-off-by: Adam Lee <ali@pivotal.io>

### CODE REVIEW

File: src/bin/gpfdist/Makefile [hunk 1]
```diff
@@ -10,17 +10,15 @@ OBJS = gpfdist.o gpfdist_helper.o fstream.o gfile.o
 # we are adding the gpfdist libraries here instead
 # of the top level, so that the backend does not
 # need to link with unnecessary libraries.
-GPFDIST_LIBS =-levent
+GPFDIST_LIBS = $(EVENT_LIBS)
 
 ifeq ($(PORTNAME),aix)
-  ifeq ($(with_libbz2),yes)
-    GPFDIST_LIBS += -lbz2
-  endif
```
Comment (danielgustafsson) [inline src/bin/gpfdist/Makefile:18]: This should be in its own commit as it's mostly unrelated to the rest of the change.

**Comment by adam8157** [comment]
Split the commit and merge.

**Comment by danielgustafsson** [comment]
> On 15 Apr 2020, at 04:01, Adam Lee <notifications@github.com> wrote:
> Split the commit and merge.
> 
Thanks for taking care of this!
